diff --git a/wrapper/compiler.h b/wrapper/compiler.h
index 0c01632..d6cbf98 100644
--- a/wrapper/compiler.h
+++ b/wrapper/compiler.h
@@ -23,20 +23,4 @@
 
 #include <linux/compiler.h>
 
-/*
- * Don't allow compiling with buggy compiler.
- */
-
-#ifdef GCC_VERSION
-
-/*
- * http://gcc.gnu.org/bugzilla/show_bug.cgi?id=58854
- */
-# ifdef __ARMEL__
-#  if GCC_VERSION >= 40800 && GCC_VERSION <= 40802
-#   error Your gcc version produces clobbered frame accesses
-#  endif
-# endif
-#endif
-
 #endif /* _LTTNG_WRAPPER_COMPILER_H */
